File: Frontend/src/types/submission.ts
import type { SubmissionStatus } from "./enums";
import type { AttachmentResponse } from "./submission-attachment";

export interface SubmissionResponse {
  id: string;
  assignmentId: string;
  assignmentTitle: string;
  studentId: string;
  studentName: string;
  studentRoll: string | null;
  answerText: string | null;
  status: SubmissionStatus;
  submittedAt: string;
  marks: number | null;
  feedback: string | null;
  evaluatedByUserId: string | null;
  evaluatedByName: string | null;
  attachments: AttachmentResponse[];
}

export interface UpsertSubmissionRequest {
  answerText?: string | null;
}

export interface ReviewSubmissionRequest {
  marks: number;
  feedback?: string | null;
  status: SubmissionStatus;
}

File: Backend/OnnorokomBackend/Models/Entities/Submission.cs
using OnnoRokomBackend.Models.Enums;

namespace OnnoRokomBackend.Models.Entities;

public class Submission
{
    public Guid Id { get; set; }
    public Guid AssignmentId { get; set; }
    public Guid StudentId { get; set; }
    public string? AnswerText { get; set; }
    public SubmissionStatus Status { get; set; } = SubmissionStatus.Submitted;
    public DateTime SubmittedAt { get; set; }
    public decimal? Marks { get; set; }
    public string? Feedback { get; set; }
    public Guid? EvaluatedByUserId { get; set; }

    public Assignment Assignment { get; set; } = null!;
    public User Student { get; set; } = null!;
    public User? EvaluatedBy { get; set; }
    public ICollection<SubmissionAttachment> Attachments { get; set; } = [];
}
